apps/Stepmania/install

```diff
@@ -0,0 +1,111 @@
+          AC_DEFINE([PNG_ARM_NEON_OPT], [0],
+                    [Disable ARM Neon optimizations])
+          # Prevent inclusion of the assembler files below:
+          enable_arm_neon=no;;
+       check)
++         AC_DEFINE([PNG_USE_ARM_NEON], [], [ARM NEON support])
+          AC_DEFINE([PNG_ARM_NEON_CHECK_SUPPORTED], [],
+                    [Check for ARM Neon support at run-time]);;
+       api)
++         AC_DEFINE([PNG_USE_ARM_NEON], [], [ARM NEON support])
+          AC_DEFINE([PNG_ARM_NEON_API_SUPPORTED], [],
+                    [Turn on ARM Neon optimizations at run-time]);;
+       yes|on)
++         AC_DEFINE([PNG_USE_ARM_NEON], [], [ARM NEON support])
+          AC_DEFINE([PNG_ARM_NEON_OPT], [2],
+                    [Enable ARM Neon optimizations])
+          AC_MSG_WARN([--enable-arm-neon: please specify 'check' or 'api', if]
+diff --git a/extern/libpng/pngpriv.h b/extern/libpng/pngpriv.h
+index 583c26f..0293a75 100644
+--- a/extern/libpng/pngpriv.h
++++ b/extern/libpng/pngpriv.h
+@@ -127,8 +127,8 @@
+     * associated assembler code, pass --enable-arm-neon=no to configure
+     * or put -DPNG_ARM_NEON_OPT=0 in CPPFLAGS.
+     */
+-#  if (defined(__ARM_NEON__) || defined(__ARM_NEON)) && \
+-   defined(PNG_ALIGNED_MEMORY_SUPPORTED)
++#  if defined(PNG_USE_ARM_NEON) && (defined(__ARM_NEON__) || \
++     defined(__ARM_NEON)) && defined(PNG_ALIGNED_MEMORY_SUPPORTED)
+ #     define PNG_ARM_NEON_OPT 2
+ #  else
+ #     define PNG_ARM_NEON_OPT 0
+diff --git a/src/arch/MovieTexture/MovieTexture_FFMpeg.h b/src/arch/MovieTexture/MovieTexture_FFMpeg.h
+index c092b76..36a6cf1 100644
+--- a/src/arch/MovieTexture/MovieTexture_FFMpeg.h
++++ b/src/arch/MovieTexture/MovieTexture_FFMpeg.h
+@@ -14,7 +14,7 @@ namespace avcodec
+ 		#include <libswscale/swscale.h>
+ 		#include <libavutil/pixdesc.h>
+ 
+-		#if LIBAVCODEC_VERSION_MAJOR >= 58
++		#if LIBAVCODEC_VERSION_MAJOR >= 57
+ 		#define av_free_packet av_packet_unref
+ 		#define PixelFormat AVPixelFormat
+ 		#define PIX_FMT_YUYV422 AV_PIX_FMT_YUYV422
+-- 
+2.25.1
+
+EOF
+
+git apply 0001-patch-upstream-for-ARM.patch || error "Applying patch failed"
+mkdir build && cd build || error "Could not create or enter build directory"
+status "Running cmake"
+cmake -DCMAKE_BUILD_TYPE=Release -DWITH_SDL=yes -DWITH_SYSTEM_GLEW=yes -DWITH_SYSTEM_TOMMATH=yes -DWITH_SYSTEM_MAD=yes -DWITH_SYSTEM_ZLIB=yes -DWITH_SYSTEM_JPEG=yes -DWITH_SYSTEM_PCRE=yes -DWITH_SYSTEM_OGG=yes -DWITH_SYSTEM_FFMPEG=yes -DWITH_FULL_RELEASE=yes -DWITH_CRASH_HANDLER=yes -DWITH_NETWORKING=yes -WITH_MINIMAID=no .. || error "Cmake ended with error"
+
+status "Starting build"
+make -j$(nproc) || "error Failed to build"
+# installs to /usr/local/stepmania-5.1 folder for everything
+status "Installing stepmania"
+sudo make install || error "Could not install"
+cd .. || error "could not move back directory"
+# install icons (the installer does not do this)
+sudo rsync -a ./icons/ /usr/local/share/icons/
```

Comment:
I noticed `rsync` was installed above. Is this the only time rsync is used? If so, could `cp` be used instead?

Reply:
Cp can't be used easily since like 10 recursive folders are copied and merged with other folders. Cp can't merge folders

Reply:
also I wouldn't worry about rsync, its a tiny dependency compared to the game itself `753.0 kB`